src/MELT/TestSink.cs

```diff
@@ -30,14 +31,28 @@ public class TestSink : ITestSink
 
         public Func<BeginScopeContext, bool>? BeginEnabled { get; set; }
 
+
         public IProducerConsumerCollection<BeginScopeContext> BeginScopes { get => _beginScopes; }
 
         public IProducerConsumerCollection<WriteContext> Writes { get => _writes; }
 
         public IEnumerable<LogEntry> LogEntries => Writes.Select(x => new LogEntry(x));
 
+        /// <summary>
+        /// All scopes that were created
+        /// </summary>
         public IEnumerable<BeginScope> Scopes => BeginScopes.Select(x => new BeginScope(x));
 
+        /// <summary>
+        /// All the current scopes.
+        /// </summary>
+        public ImmutableStack<object?> CurrentScopeData => _currentScope.ScopeData;
```

Comment:
I don't think exposing the scopes list as immutable adds much value in this context, I would probably keep it simple an use IEnumerable like the rest of the data.
Using object for the data doesn't feel user friendly, can we not use `BeginScope` here as well? or we need something different?

Reply:
Okay. I've changed it to IEnumerable<IScope> and done the same thing in LogEntry